ensime_shared/client.py

```diff
@@ -224,26 +202,10 @@ def tell_module_missing(self, name):
         msg = feedback["module_missing"]
         self.editor.raw_message(msg.format(name, name))
 
-    # TODO: this should be the Ensime class's responsibility, not EnsimeClient;
-    #       move (and eliminate self.vim) after fixing, see #294
-    def disable_plugin(self):
-        """Disable plugin temporarily, including also related plugins."""
-        self.log.debug('disable_plugin: in')
-
-        def threadsafe_vim(command):
-            """Threadsafe call if neovim, normal if vim."""
-            def normal_vim(e):
-                self.vim.command(command)
-            with catch(Exception, normal_vim):
-                self.vim.session.threadsafe_call(command)
```

Comment:
This was actually broken, we couldn't tell because of catching `Exception` 😖 This method takes a function as argument, not a string of a Vim command. `vim.async_call` is a higher-level wrapper for it in the Neovim Python API.